### ISSUE TITLE
Catch BindException on SimpleServer.startAndWait()

### ISSUE THREAD

**Issue by zaewonyx** [hapifhir/hapi-hl7v2#8]
I'm trying to catch the binding exception when the MLLP Server is started on a port already in use, in order to react to this error, but I'm affraid that is currently impossible.

Looking at the code, in SimpleServer, the `startupLatch.await()` is executed just after the `start()` method of the `Service` class but, as the `AcceptorThread` is launch in an other thread (by definition), the `startupLatch.countDown()` of the SimpleServer is called before the end of execution of the `afterStartup()` of the acceptor thread (which is causing the BindException).

In other words, in a normal context, if i write this wanted snippet :

`    
  
    this.simpleServer.startAndWait();
    
    Throwable startingException = this.simpleServer.getServiceExitedWithException();
    if (startingException != null) {
	// do something
    }`

The `getServiceExitedWithException()` is always null. The only solution for now is to wait for x ms and in this case the exception is filled.

Am I missing something or is there an other way to do it ?

Thanks

**Comment by yuflyud** [comment]
I have the same issue, my integration tests fail from time to time.
